tests/test_read_lazy.py

```diff
@@ -0,0 +1,592 @@
+    concated_remote = ad.concat(lazy_adatas, join=join)
+    # check preservation of non-categorical dtypes on the concat axis
+    assert concated_remote.obs["int64"].dtype == "int64"
+    assert concated_remote.obs["uint8"].dtype == "uint8"
+    assert concated_remote.obs["nullable-int"].dtype == "int32"
+    assert concated_remote.obs["float64"].dtype == "float64"
+    assert concated_remote.obs["bool"].dtype == "bool"
+    assert concated_remote.obs["nullable-bool"].dtype == "bool"
+
+
+def test_concat_to_memory_var(
+    concatenation_objects: tuple[
+        list[AnnData], list[pd.Index], list[AccessTrackingStore], list[AnnData]
+    ],
+    join: Join_T,
+    are_vars_different: bool,
+    simple_subset_func: Callable[[AnnData], AnnData],
+):
+    adatas, var_indices, stores, lazy_adatas = concatenation_objects
+    concated_remote = simple_subset_func(ad.concat(lazy_adatas, join=join))
+    var_keys_to_track = get_key_trackers_for_columns_on_axis(adatas[0], "var")
+    for store in stores:
+        store.initialize_key_trackers(var_keys_to_track)
+    # check non-different variables, taken from first annotation.
+    pd_index_overlapping = pd.Index(
+        filter(lambda x: not x.endswith("ds"), var_indices[0])
+    )
+    var_df_overlapping = adatas[0][:, pd_index_overlapping].var.copy()
+    test_cases = [(pd_index_overlapping, var_df_overlapping, 0)]
+    if are_vars_different and join == "outer":
+        # check a set of unique variables from the first object since we only take from there if different
+        pd_index_only_ds_0 = pd.Index(filter(lambda x: "0_ds" in x, var_indices[1]))
+        var_df_only_ds_0 = adatas[0][:, pd_index_only_ds_0].var.copy()
+        test_cases.append((pd_index_only_ds_0, var_df_only_ds_0, 0))
+    for pd_index, var_df, store_idx in test_cases:
+        var_df.index.name = "var_names"
+        remote_df = concated_remote[:, pd_index].var.to_pandas()
+        remote_df_corrected, _ = correct_extension_dtype_differences(remote_df, var_df)
+        #  TODO:xr.merge always upcasts to float due to NA and you can't downcast?
+        for col in remote_df_corrected.columns:
+            dtype = remote_df_corrected[col].dtype
+            if dtype in [np.float64, np.float32]:
+                var_df[col] = var_df[col].astype(dtype)
+        assert_equal(remote_df_corrected, var_df)
+        for key in var_keys_to_track:
+            stores[store_idx].assert_access_count(key, 1)
+            for store in stores:
+                if store != stores[store_idx]:
+                    store.assert_access_count(key, 0)
+        stores[store_idx].reset_key_trackers()
+
+
+@pytest.mark.parametrize(
+    "index",
+    [
+        pytest.param(
+            slice(500, 1500),
+            id="slice",
+        ),
+        pytest.param(
+            np.arange(950, 1050),
+            id="consecutive integer array",
+        ),
+        pytest.param(
+            np.random.choice(np.arange(800, 1100), 500),
```

Comment:
isn’t that [randint](https://numpy.org/doc/stable/reference/random/generated/numpy.random.randint.html)?